Adding tag support

AlienHoboken opened the pull request:
Added tag support for DO Api, including listing and deleting droplets by tag.

vevix commented on DigitalOcean.API/Clients/TagsClient.cs:
```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        public Task<Tag> Create(string tagName) {
+            Models.Requests.Tag data = new Models.Requests.Tag();
+            data.Name = tagName;
```
Swap this for an Object Initializer

```csharp
var data = new Models.Requests.Tag {
    Name = tagName
};
```

vevix commented on DigitalOcean.API/Clients/TagsClient.cs:
```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        public Task<Tag> Update(string tagName, string newTagName) {
+            Models.Requests.Tag data = new Models.Requests.Tag();
+            data.Name = newTagName;
```
Swap this for an Object Initializer

vevix commented on DigitalOcean.API/Clients/TagsClient.cs:
```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        public Task Tag(string tagName, List<KeyValuePair<string, string>> resources) {
+            Models.Requests.TagResource data = new Models.Requests.TagResource();
+            data.Resources = new List<Models.Requests.TagResource.Resource>();
```
Swap this for an Object Initializer

vevix commented on DigitalOcean.API/Clients/TagsClient.cs:
```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        public Task Untag(string tagName, List<KeyValuePair<string, string>> resources) {
+            Models.Requests.TagResource data = new Models.Requests.TagResource();
+            data.Resources = new List<Models.Requests.TagResource.Resource>();
```
Swap this for an Object Initializer

vevix commented:
Great work! :100:

Could you go thru and fix the style issues so the codebase is consistent? I left some comments, but there is also a few mismatching opening brackets (Codebase follows [1TBS](https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS_.28OTBS.29))

AlienHoboken commented:
Missed those object initializers and a few brackets, thanks for pointing it out! Fixed the style issues, even found a couple that weren't mine and corrected. 😊

vevix commented:
Thanks! I'll merge this when I'm ready to push a new version to NuGet. Hopefully tonight.
